### PR TITLE
Extend AES sideload test

### PR DESCRIPTION
**Pull request by gdessouky**
Extend the AES sideload test to disable the continuous clearing and to generate the same key again after clearing.

Signed-off-by: Ghada Dessouky  <gdessouky@google.com>

### CODE REVIEW

**Comment by gdessouky** [comment]
@weicaiyang I extend here the AES sideload test to disable the continuous clearing and to generate the same key again to also test that out for AES.

File: sw/device/tests/keymgr_sideload_aes_test.c [hunk 1]
```diff
@@ -153,6 +159,33 @@ void aes_test(void) {
...
+
+  //  Verify that the ciphertext is identical to the first one generated.
+  dif_aes_data_t out_data_second;
+  CHECK_DIF_OK(dif_aes_read_output(&aes, &out_data_second));
+  CHECK_ARRAYS_EQ(out_data.data, out_data_second.data,
+                  ARRAYSIZE(out_data.data));
+
+  // Finish the ECB encryption transaction.
+  LOG_INFO("Finished encrypting again with the same key");
+  CHECK_DIF_OK(dif_aes_end(&aes));
```
Comment (weicaiyang) [inline sw/device/tests/keymgr_sideload_aes_test.c:188]: looks like this encrypt operation is replicated in 3 places. can we make it a function? perhaps `out_data` can be the function output and we can check it in `aes_test`

**Review by weicaiyang — APPROVED** [review]
LGTM with a suggestion

File: sw/device/tests/keymgr_sideload_aes_test.c (hunk 1 above)
Comment (gdessouky) [reply sw/device/tests/keymgr_sideload_aes_test.c:188]: yes, that makes sense. thanks, will take care of it.

Comment (gdessouky) [reply sw/device/tests/keymgr_sideload_aes_test.c]: @weicaiyang I moved the encrypt/decrypt operations into one function and call it as you suggest. Let me know if it looks good now.

**Review by weicaiyang — APPROVED** [review]

**Pull request merged**